packages/cubejs-client-core/dist/index.d.ts

```diff
@@ -82,24 +97,36 @@ export type Query = {
   limit?: number;
   offset?: number;
   order?: {
-    [key: string]: QueryOrderOptions;
+    [key: string]: QueryOrder;
   };
   timezone?: string;
   renewQuery?: boolean;
   ungrouped?: boolean;
 };
 
+export type ProgressResponse = {
+  stage: string;
+  timeElapsed: number;
+}
+
+export class ProgressResult {
+  new(progressResponse: ProgressResponse): ProgressResult;
+
+  stage(): string;
+  timeElapsed(): string;
+}
+
 export class CubejsApi {
   new(apiToken: string, options: CubeJSApiOptions): CubejsApi;
 
-  load(query: Query, options, callback): void;
-  load(query: Query, options): Promise<ResultSet>;
+  load(query: Query, options?: LoadMethodOptions, callback?: LoadMethodCallback): void;
+  load(query: Query, options?: LoadMethodOptions): Promise<ResultSet>;
 
-  sql(query: Query, options, callback): void;
-  sql(query: Query, options): Promise<ResultSet>;
+  sql(query: Query, options?: LoadMethodOptions, callback?: LoadMethodCallback): void;
+  sql(query: Query, options?: LoadMethodOptions): Promise<ResultSet>;
```

Comment:
`ResultSet` should be replaced here with something

Reply:
Hi @paveltiunov, not quite sure what you mean by that? What should `ResultSet` be replaced by?

Reply:
@hassankhan `sql()` method returns `SqlQuery` instances of class and not `ResultSet` with query results. Same for `meta()`: it returns `Meta` instances of class.

Reply:
@paveltiunov Added types for `SqlQuery`